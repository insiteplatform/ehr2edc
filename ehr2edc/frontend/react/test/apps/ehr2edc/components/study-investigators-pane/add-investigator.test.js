import React from 'react';
import renderer from 'react-test-renderer';
import AddInvestigator from "../../../../../src/apps/ehr2edc/components/study-investigators-pane/add-investigator";

'use strict';
const fs = require('fs');
const path = require("path");

const webserviceResponseFolder = "../../../../../../../infrastructure/web/src/test/resources/samples/";
const potentialInvestigatorResponse = readData("userscontroller/unassigned-response");

function readData(filePath) {
    const fileContents = fs.readFileSync(path.resolve(__dirname, webserviceResponseFolder + filePath + ".json"), 'utf-8');
    return JSON.parse(fileContents);
}

describe("Component gets rendered well", () => {
    test("The component is closed by default", () => {
        const app = renderer.create(<AddInvestigator
            potentialInvestigators={potentialInvestigatorResponse.potentialInvestigators}/>);
        const tree = app.toJSON();
        expect(tree).toMatchSnapshot();
    });
    test("Closed state gets rendered well", () => {
        const app = renderer.create(<AddInvestigator
            potentialInvestigators={potentialInvestigatorResponse.potentialInvestigators}
            isOpen={false}/>);
        const tree = app.toJSON();
        expect(tree).toMatchSnapshot();
    });
    test("Open state gets rendered well", () => {
        const app = renderer.create(<AddInvestigator
            potentialInvestigators={potentialInvestigatorResponse.potentialInvestigators}
            isOpen={true}/>);
        const tree = app.toJSON();
        expect(tree).toMatchSnapshot();
    });
    test("The empty state is rendered when no investigators are passed", () => {
        const app = renderer.create(<AddInvestigator/>);
        const tree = app.toJSON();
        expect(tree).toMatchSnapshot();
    });
    test("The empty state is rendered when an empty list is passed", () => {
        const app = renderer.create(<AddInvestigator potentialInvestigators={[]}/>);
        const tree = app.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
describe("Switching between the open and closed state manually", () => {
    test("Clicking the add button in the closed state switches to the open state", async () => {
        jest.useFakeTimers();
        const app = mount(<AddInvestigator potentialInvestigators={potentialInvestigatorResponse.potentialInvestigators}
                                           isOpen={false}/>);
        await Promise.resolve();
        app.find('.open-add-investigator').simulate('click', {});
        app.update();
        expect(app).toMatchSnapshot();
    });
    test("Clicking the cancel button in the open state switches to the closed state", async () => {
        jest.useFakeTimers();
        const app = mount(<AddInvestigator potentialInvestigators={potentialInvestigatorResponse.potentialInvestigators}
                                           isOpen={true}/>);
        await Promise.resolve();
        app.find('.close-add-investigator').simulate('click', {});
        app.update();
        expect(app).toMatchSnapshot();
    });
});
describe("Interactions with the form", () => {
    let onInvestigatorAdd = jest.fn();

    beforeEach(() => {
        onInvestigatorAdd.mockReset();
    });

    async function getAddInvestigatorPanelWithData() {
        jest.useFakeTimers();
        return mount(<AddInvestigator potentialInvestigators={potentialInvestigatorResponse.potentialInvestigators}
                                      onInvestigatorAdd={onInvestigatorAdd}
                                      isOpen={true}/>);
    }

    function focusAndOpenAutocomplete(app) {
        app.find('.autocomplete-input').simulate('focus', {});
        document.dispatchEvent(new KeyboardEvent('keydown', {keyCode: 40}));
        app.update();
    }

    function findAutocompleteItem(app, itemKey) {
        return app.findWhere(item => item.key() === itemKey);
    }

    test("The autocomplete component doesn't show anything when no interactions have happened", async () => {
        const app = await getAddInvestigatorPanelWithData();
        app.update();
        expect(app).toMatchSnapshot();
    });
    test("Giving the autocomplete box focus opens the full list of investigators that can be added", async () => {
        const app = await getAddInvestigatorPanelWithData();
        app.find('.autocomplete-input').simulate('focus', {});
        await Promise.resolve();
        app.update();
        expect(app).toMatchSnapshot();
    });
    test("Taking away the autocomplete box's focus hides the list of investigators that can be added", async () => {
        const app = await getAddInvestigatorPanelWithData();
        app.find('.autocomplete-input').simulate('focus', {});
        app.find('.autocomplete-input').simulate('blur', {});
        await Promise.resolve();
        app.update();
        expect(app).toMatchSnapshot();
    });
    test("Typing in the autocomplete box filters the list of investigators that can be added", async () => {
        const app = await getAddInvestigatorPanelWithData();
        app.find('.autocomplete-input').simulate('change', {target: {value: "1"}});
        app.find('.autocomplete-input').simulate('focus', {});
        await Promise.resolve();
        app.update();
        expect(app).toMatchSnapshot();
    });
    test("Hovering over an item in the autocomplete menu adds an highlight to it", async () => {
        const app = await getAddInvestigatorPanelWithData();
        focusAndOpenAutocomplete(app);
        findAutocompleteItem(app, 'User id. 7').simulate('mouseenter', {});
        await Promise.resolve();
        app.update();
        expect(app.findWhere(item => item.key() === 'User id. 7').hasClass("highlighted")).toBe(true)
    });
    test("Clicking an item in the autocomplete menu fills it into the autocomplete element", async () => {
        const app = await getAddInvestigatorPanelWithData();
        focusAndOpenAutocomplete(app);
        findAutocompleteItem(app, 'User id. 3').simulate('click', {});
        await Promise.resolve();
        app.update();
        expect(app.find('.autocomplete-input').getDOMNode().value).toEqual("User nr. 3");
    });
    test("Submitting the form with valid input", async () => {
        // Given: A selected potential investigator
        const spyOnSubmit = jest.spyOn(AddInvestigator.prototype, 'handleSubmit');
        const app = await getAddInvestigatorPanelWithData();
        focusAndOpenAutocomplete(app);
        findAutocompleteItem(app, 'User id. 3').simulate('click', {});

        // When: Submitting the form
        app.find('form').simulate('submit');

        // Then: The handler was called
        expect(spyOnSubmit).toHaveBeenCalled();
        // And: The callback was triggered
        expect(onInvestigatorAdd).toHaveBeenCalledWith({
            id: 'User id. 3',
            name: 'User nr. 3'
        })
    });
    test("Submitting the form with invalid input", async () => {
        // Given: A selected potential investigator
        const spyOnSubmit = jest.spyOn(AddInvestigator.prototype, 'handleSubmit');
        const app = await getAddInvestigatorPanelWithData();
        focusAndOpenAutocomplete(app);

        // When: Submitting the form
        app.find('form').simulate('submit');

        // Then: The handler was called
        expect(spyOnSubmit).toHaveBeenCalled();
        // And: The callback was triggered
        expect(onInvestigatorAdd).not.toHaveBeenCalled()
    });
    test("Closing the form with the 'Esc'-key", async () => {
        // Given: A form with opened autocomplete
        const app = await getAddInvestigatorPanelWithData();
        focusAndOpenAutocomplete(app);

        // When: Pressing Esc
        document.dispatchEvent(new KeyboardEvent('keydown', {keyCode: 27}));
        app.update();

        // Then: The form was closed
        expect(app.find("form").exists()).toBe(false);
    });
});
describe("Clicking the add button", () => {
    function selectInvestigator(app) {
        app.find('.autocomplete-input').simulate('focus', {});
        document.dispatchEvent(new KeyboardEvent('keydown', {keyCode: 40}));
        app.update();
        app.findWhere(item => item.key() === 'User id. 3').simulate('click', {});
    }

    test("Clicking the add button without selecting an investigator does nothing", async () => {
        jest.useFakeTimers();
        let onInvestigatorAdd = jest.fn();
        const app = mount(<AddInvestigator potentialInvestigators={potentialInvestigatorResponse.potentialInvestigators}
                                           isOpen={true}
                                           onInvestigatorAdd={onInvestigatorAdd}/>);

        app.find(".add-investigator").simulate('click', {});

        expect(onInvestigatorAdd.mock.calls.length).toEqual(0);
    });
    test("Clicking the add button after selecting an investigator fires an event", async () => {
        jest.useFakeTimers();
        let onInvestigatorAdd = jest.fn();
        const app = mount(<AddInvestigator potentialInvestigators={potentialInvestigatorResponse.potentialInvestigators}
                                           isOpen={true}
                                           onInvestigatorAdd={onInvestigatorAdd}/>);

        selectInvestigator(app);
        app.find(".add-investigator").simulate('click', {});

        expect(onInvestigatorAdd.mock.calls.length).toEqual(1);
    });
    test("Clicking the add button after selecting an investigator passes the correct investigator", async () => {
        jest.useFakeTimers();
        let onInvestigatorAdd = jest.fn();
        const app = mount(<AddInvestigator potentialInvestigators={potentialInvestigatorResponse.potentialInvestigators}
                                           isOpen={true}
                                           onInvestigatorAdd={onInvestigatorAdd}/>);

        selectInvestigator(app);
        app.find(".add-investigator").simulate('click', {});

        expect(onInvestigatorAdd).toHaveBeenCalledWith({"id": "User id. 3", "name": "User nr. 3"});
    });
});