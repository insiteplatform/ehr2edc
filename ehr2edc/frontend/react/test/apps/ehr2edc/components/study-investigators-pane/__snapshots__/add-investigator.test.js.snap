// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component gets rendered well Closed state gets rendered well 1`] = `
<div
  className="grid-content collapse shrink"
>
  <a
    className="button open-add-investigator"
    onClick={[Function]}
  >
    <i
      className="fa fa-plus"
    />
     Add investigator
  </a>
</div>
`;

exports[`Component gets rendered well Open state gets rendered well 1`] = `
<div
  className="grid-content collapse shrink"
>
  <form
    onSubmit={[Function]}
  >
    <span
      className="inline-label"
    >
      <div
        className="autocomplete-override"
      >
        <div
          style={
            Object {
              "display": "inline-block",
            }
          }
        >
          <input
            aria-autocomplete="list"
            aria-expanded={false}
            autoComplete="off"
            autoFocus={true}
            className="autocomplete-input margin-none"
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Add investigator ..."
            role="combobox"
            type="text"
            value=""
          />
        </div>
      </div>
      <a
        className="button disabled add-investigator"
      >
        <i
          className="fa fa-plus margin-small-right"
        />
         Add
      </a>
      <a
        className="button secondary close-add-investigator"
        onClick={[Function]}
      >
        <i
          className="fa fa-times margin-small-right"
        />
        Cancel
      </a>
    </span>
  </form>
</div>
`;

exports[`Component gets rendered well The component is closed by default 1`] = `
<div
  className="grid-content collapse shrink"
>
  <a
    className="button open-add-investigator"
    onClick={[Function]}
  >
    <i
      className="fa fa-plus"
    />
     Add investigator
  </a>
</div>
`;

exports[`Component gets rendered well The empty state is rendered when an empty list is passed 1`] = `
<div
  className="grid-content collapse shrink"
  title="All available users have been assigned as investigator"
>
  <a
    className="button open-add-investigator disabled"
  >
    <i
      className="fa fa-plus"
    />
     Add investigator
  </a>
</div>
`;

exports[`Component gets rendered well The empty state is rendered when no investigators are passed 1`] = `
<div
  className="grid-content collapse shrink"
  title="All available users have been assigned as investigator"
>
  <a
    className="button open-add-investigator disabled"
  >
    <i
      className="fa fa-plus"
    />
     Add investigator
  </a>
</div>
`;

exports[`Interactions with the form Giving the autocomplete box focus opens the full list of investigators that can be added 1`] = `
<AddInvestigator
  isOpen={true}
  onInvestigatorAdd={[MockFunction]}
  potentialInvestigators={
    Array [
      Object {
        "id": "User id. 1",
        "name": "User nr. 1",
      },
      Object {
        "id": "User id. 2",
        "name": "User nr. 2",
      },
      Object {
        "id": "User id. 3",
        "name": "User nr. 3",
      },
      Object {
        "id": "User id. 4",
        "name": "User nr. 4",
      },
      Object {
        "id": "User id. 5",
        "name": "User nr. 5",
      },
      Object {
        "id": "User id. 6",
        "name": "User nr. 6",
      },
      Object {
        "id": "User id. 7",
        "name": "User nr. 7",
      },
      Object {
        "id": "User id. 8",
        "name": "User nr. 8",
      },
      Object {
        "id": "User id. 9",
        "name": "User nr. 9",
      },
      Object {
        "id": "User id. 10",
        "name": "User nr. 10",
      },
    ]
  }
>
  <div
    className="grid-content collapse shrink"
  >
    <form
      onSubmit={[Function]}
    >
      <span
        className="inline-label"
      >
        <AutocompleteInput
          alwaysRender={false}
          inputProps={
            Object {
              "autoFocus": true,
            }
          }
          onSelectionChange={[Function]}
          openOnFocus={false}
          options={
            Array [
              Object {
                "label": "User nr. 1",
                "value": "User id. 1",
              },
              Object {
                "label": "User nr. 2",
                "value": "User id. 2",
              },
              Object {
                "label": "User nr. 3",
                "value": "User id. 3",
              },
              Object {
                "label": "User nr. 4",
                "value": "User id. 4",
              },
              Object {
                "label": "User nr. 5",
                "value": "User id. 5",
              },
              Object {
                "label": "User nr. 6",
                "value": "User id. 6",
              },
              Object {
                "label": "User nr. 7",
                "value": "User id. 7",
              },
              Object {
                "label": "User nr. 8",
                "value": "User id. 8",
              },
              Object {
                "label": "User nr. 9",
                "value": "User id. 9",
              },
              Object {
                "label": "User nr. 10",
                "value": "User id. 10",
              },
            ]
          }
          placeholder="Add investigator ..."
        >
          <div
            className="autocomplete-override"
          >
            <Autocomplete
              autoHighlight={true}
              getItemValue={[Function]}
              inputProps={
                Object {
                  "autoFocus": true,
                }
              }
              isItemSelectable={[Function]}
              items={Array []}
              menuStyle={
                Object {
                  "background": "rgba(255, 255, 255, 0.9)",
                  "borderRadius": "3px",
                  "boxShadow": "0 2px 12px rgba(0, 0, 0, 0.1)",
                  "fontSize": "90%",
                  "maxHeight": "50%",
                  "overflow": "auto",
                  "padding": "2px 0",
                  "position": "fixed",
                }
              }
              onChange={[Function]}
              onMenuVisibilityChange={[Function]}
              onSelect={[Function]}
              renderInput={[Function]}
              renderItem={[Function]}
              renderMenu={[Function]}
              selectOnBlur={false}
              shouldItemRender={[Function]}
              value=""
              wrapperProps={Object {}}
              wrapperStyle={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <div
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-autocomplete="list"
                  aria-expanded={true}
                  autoComplete="off"
                  autoFocus={true}
                  className="autocomplete-input margin-none"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Add investigator ..."
                  role="combobox"
                  type="text"
                  value=""
                />
                <div
                  className="autocomplete-menu"
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onTouchStart={[Function]}
                  style={
                    Object {
                      "left": 0,
                      "minWidth": 0,
                      "top": 0,
                    }
                  }
                />
              </div>
            </Autocomplete>
          </div>
        </AutocompleteInput>
        <a
          className="button disabled add-investigator"
        >
          <i
            className="fa fa-plus margin-small-right"
          />
           Add
        </a>
        <a
          className="button secondary close-add-investigator"
          onClick={[Function]}
        >
          <i
            className="fa fa-times margin-small-right"
          />
          Cancel
        </a>
      </span>
    </form>
  </div>
</AddInvestigator>
`;

exports[`Interactions with the form Taking away the autocomplete box's focus hides the list of investigators that can be added 1`] = `
<AddInvestigator
  isOpen={true}
  onInvestigatorAdd={[MockFunction]}
  potentialInvestigators={
    Array [
      Object {
        "id": "User id. 1",
        "name": "User nr. 1",
      },
      Object {
        "id": "User id. 2",
        "name": "User nr. 2",
      },
      Object {
        "id": "User id. 3",
        "name": "User nr. 3",
      },
      Object {
        "id": "User id. 4",
        "name": "User nr. 4",
      },
      Object {
        "id": "User id. 5",
        "name": "User nr. 5",
      },
      Object {
        "id": "User id. 6",
        "name": "User nr. 6",
      },
      Object {
        "id": "User id. 7",
        "name": "User nr. 7",
      },
      Object {
        "id": "User id. 8",
        "name": "User nr. 8",
      },
      Object {
        "id": "User id. 9",
        "name": "User nr. 9",
      },
      Object {
        "id": "User id. 10",
        "name": "User nr. 10",
      },
    ]
  }
>
  <div
    className="grid-content collapse shrink"
  >
    <form
      onSubmit={[Function]}
    >
      <span
        className="inline-label"
      >
        <AutocompleteInput
          alwaysRender={false}
          inputProps={
            Object {
              "autoFocus": true,
            }
          }
          onSelectionChange={[Function]}
          openOnFocus={false}
          options={
            Array [
              Object {
                "label": "User nr. 1",
                "value": "User id. 1",
              },
              Object {
                "label": "User nr. 2",
                "value": "User id. 2",
              },
              Object {
                "label": "User nr. 3",
                "value": "User id. 3",
              },
              Object {
                "label": "User nr. 4",
                "value": "User id. 4",
              },
              Object {
                "label": "User nr. 5",
                "value": "User id. 5",
              },
              Object {
                "label": "User nr. 6",
                "value": "User id. 6",
              },
              Object {
                "label": "User nr. 7",
                "value": "User id. 7",
              },
              Object {
                "label": "User nr. 8",
                "value": "User id. 8",
              },
              Object {
                "label": "User nr. 9",
                "value": "User id. 9",
              },
              Object {
                "label": "User nr. 10",
                "value": "User id. 10",
              },
            ]
          }
          placeholder="Add investigator ..."
        >
          <div
            className="autocomplete-override"
          >
            <Autocomplete
              autoHighlight={true}
              getItemValue={[Function]}
              inputProps={
                Object {
                  "autoFocus": true,
                }
              }
              isItemSelectable={[Function]}
              items={Array []}
              menuStyle={
                Object {
                  "background": "rgba(255, 255, 255, 0.9)",
                  "borderRadius": "3px",
                  "boxShadow": "0 2px 12px rgba(0, 0, 0, 0.1)",
                  "fontSize": "90%",
                  "maxHeight": "50%",
                  "overflow": "auto",
                  "padding": "2px 0",
                  "position": "fixed",
                }
              }
              onChange={[Function]}
              onMenuVisibilityChange={[Function]}
              onSelect={[Function]}
              renderInput={[Function]}
              renderItem={[Function]}
              renderMenu={[Function]}
              selectOnBlur={false}
              shouldItemRender={[Function]}
              value=""
              wrapperProps={Object {}}
              wrapperStyle={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <div
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-autocomplete="list"
                  aria-expanded={false}
                  autoComplete="off"
                  autoFocus={true}
                  className="autocomplete-input margin-none"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Add investigator ..."
                  role="combobox"
                  type="text"
                  value=""
                />
              </div>
            </Autocomplete>
          </div>
        </AutocompleteInput>
        <a
          className="button disabled add-investigator"
        >
          <i
            className="fa fa-plus margin-small-right"
          />
           Add
        </a>
        <a
          className="button secondary close-add-investigator"
          onClick={[Function]}
        >
          <i
            className="fa fa-times margin-small-right"
          />
          Cancel
        </a>
      </span>
    </form>
  </div>
</AddInvestigator>
`;

exports[`Interactions with the form The autocomplete component doesn't show anything when no interactions have happened 1`] = `
<AddInvestigator
  isOpen={true}
  onInvestigatorAdd={[MockFunction]}
  potentialInvestigators={
    Array [
      Object {
        "id": "User id. 1",
        "name": "User nr. 1",
      },
      Object {
        "id": "User id. 2",
        "name": "User nr. 2",
      },
      Object {
        "id": "User id. 3",
        "name": "User nr. 3",
      },
      Object {
        "id": "User id. 4",
        "name": "User nr. 4",
      },
      Object {
        "id": "User id. 5",
        "name": "User nr. 5",
      },
      Object {
        "id": "User id. 6",
        "name": "User nr. 6",
      },
      Object {
        "id": "User id. 7",
        "name": "User nr. 7",
      },
      Object {
        "id": "User id. 8",
        "name": "User nr. 8",
      },
      Object {
        "id": "User id. 9",
        "name": "User nr. 9",
      },
      Object {
        "id": "User id. 10",
        "name": "User nr. 10",
      },
    ]
  }
>
  <div
    className="grid-content collapse shrink"
  >
    <form
      onSubmit={[Function]}
    >
      <span
        className="inline-label"
      >
        <AutocompleteInput
          alwaysRender={false}
          inputProps={
            Object {
              "autoFocus": true,
            }
          }
          onSelectionChange={[Function]}
          openOnFocus={false}
          options={
            Array [
              Object {
                "label": "User nr. 1",
                "value": "User id. 1",
              },
              Object {
                "label": "User nr. 2",
                "value": "User id. 2",
              },
              Object {
                "label": "User nr. 3",
                "value": "User id. 3",
              },
              Object {
                "label": "User nr. 4",
                "value": "User id. 4",
              },
              Object {
                "label": "User nr. 5",
                "value": "User id. 5",
              },
              Object {
                "label": "User nr. 6",
                "value": "User id. 6",
              },
              Object {
                "label": "User nr. 7",
                "value": "User id. 7",
              },
              Object {
                "label": "User nr. 8",
                "value": "User id. 8",
              },
              Object {
                "label": "User nr. 9",
                "value": "User id. 9",
              },
              Object {
                "label": "User nr. 10",
                "value": "User id. 10",
              },
            ]
          }
          placeholder="Add investigator ..."
        >
          <div
            className="autocomplete-override"
          >
            <Autocomplete
              autoHighlight={true}
              getItemValue={[Function]}
              inputProps={
                Object {
                  "autoFocus": true,
                }
              }
              isItemSelectable={[Function]}
              items={Array []}
              menuStyle={
                Object {
                  "background": "rgba(255, 255, 255, 0.9)",
                  "borderRadius": "3px",
                  "boxShadow": "0 2px 12px rgba(0, 0, 0, 0.1)",
                  "fontSize": "90%",
                  "maxHeight": "50%",
                  "overflow": "auto",
                  "padding": "2px 0",
                  "position": "fixed",
                }
              }
              onChange={[Function]}
              onMenuVisibilityChange={[Function]}
              onSelect={[Function]}
              renderInput={[Function]}
              renderItem={[Function]}
              renderMenu={[Function]}
              selectOnBlur={false}
              shouldItemRender={[Function]}
              value=""
              wrapperProps={Object {}}
              wrapperStyle={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <div
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-autocomplete="list"
                  aria-expanded={false}
                  autoComplete="off"
                  autoFocus={true}
                  className="autocomplete-input margin-none"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Add investigator ..."
                  role="combobox"
                  type="text"
                  value=""
                />
              </div>
            </Autocomplete>
          </div>
        </AutocompleteInput>
        <a
          className="button disabled add-investigator"
        >
          <i
            className="fa fa-plus margin-small-right"
          />
           Add
        </a>
        <a
          className="button secondary close-add-investigator"
          onClick={[Function]}
        >
          <i
            className="fa fa-times margin-small-right"
          />
          Cancel
        </a>
      </span>
    </form>
  </div>
</AddInvestigator>
`;

exports[`Interactions with the form Typing in the autocomplete box filters the list of investigators that can be added 1`] = `
<AddInvestigator
  isOpen={true}
  onInvestigatorAdd={[MockFunction]}
  potentialInvestigators={
    Array [
      Object {
        "id": "User id. 1",
        "name": "User nr. 1",
      },
      Object {
        "id": "User id. 2",
        "name": "User nr. 2",
      },
      Object {
        "id": "User id. 3",
        "name": "User nr. 3",
      },
      Object {
        "id": "User id. 4",
        "name": "User nr. 4",
      },
      Object {
        "id": "User id. 5",
        "name": "User nr. 5",
      },
      Object {
        "id": "User id. 6",
        "name": "User nr. 6",
      },
      Object {
        "id": "User id. 7",
        "name": "User nr. 7",
      },
      Object {
        "id": "User id. 8",
        "name": "User nr. 8",
      },
      Object {
        "id": "User id. 9",
        "name": "User nr. 9",
      },
      Object {
        "id": "User id. 10",
        "name": "User nr. 10",
      },
    ]
  }
>
  <div
    className="grid-content collapse shrink"
  >
    <form
      onSubmit={[Function]}
    >
      <span
        className="inline-label"
      >
        <AutocompleteInput
          alwaysRender={false}
          inputProps={
            Object {
              "autoFocus": true,
            }
          }
          onSelectionChange={[Function]}
          openOnFocus={false}
          options={
            Array [
              Object {
                "label": "User nr. 1",
                "value": "User id. 1",
              },
              Object {
                "label": "User nr. 2",
                "value": "User id. 2",
              },
              Object {
                "label": "User nr. 3",
                "value": "User id. 3",
              },
              Object {
                "label": "User nr. 4",
                "value": "User id. 4",
              },
              Object {
                "label": "User nr. 5",
                "value": "User id. 5",
              },
              Object {
                "label": "User nr. 6",
                "value": "User id. 6",
              },
              Object {
                "label": "User nr. 7",
                "value": "User id. 7",
              },
              Object {
                "label": "User nr. 8",
                "value": "User id. 8",
              },
              Object {
                "label": "User nr. 9",
                "value": "User id. 9",
              },
              Object {
                "label": "User nr. 10",
                "value": "User id. 10",
              },
            ]
          }
          placeholder="Add investigator ..."
        >
          <div
            className="autocomplete-override"
          >
            <Autocomplete
              autoHighlight={true}
              getItemValue={[Function]}
              inputProps={
                Object {
                  "autoFocus": true,
                }
              }
              isItemSelectable={[Function]}
              items={Array []}
              menuStyle={
                Object {
                  "background": "rgba(255, 255, 255, 0.9)",
                  "borderRadius": "3px",
                  "boxShadow": "0 2px 12px rgba(0, 0, 0, 0.1)",
                  "fontSize": "90%",
                  "maxHeight": "50%",
                  "overflow": "auto",
                  "padding": "2px 0",
                  "position": "fixed",
                }
              }
              onChange={[Function]}
              onMenuVisibilityChange={[Function]}
              onSelect={[Function]}
              renderInput={[Function]}
              renderItem={[Function]}
              renderMenu={[Function]}
              selectOnBlur={false}
              shouldItemRender={[Function]}
              value="1"
              wrapperProps={Object {}}
              wrapperStyle={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <div
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-autocomplete="list"
                  aria-expanded={true}
                  autoComplete="off"
                  autoFocus={true}
                  className="autocomplete-input margin-none"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Add investigator ..."
                  role="combobox"
                  type="text"
                  value="1"
                />
                <div
                  className="autocomplete-menu"
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onTouchStart={[Function]}
                  style={
                    Object {
                      "left": 0,
                      "minWidth": 0,
                      "top": 0,
                    }
                  }
                />
              </div>
            </Autocomplete>
          </div>
        </AutocompleteInput>
        <a
          className="button disabled add-investigator"
        >
          <i
            className="fa fa-plus margin-small-right"
          />
           Add
        </a>
        <a
          className="button secondary close-add-investigator"
          onClick={[Function]}
        >
          <i
            className="fa fa-times margin-small-right"
          />
          Cancel
        </a>
      </span>
    </form>
  </div>
</AddInvestigator>
`;

exports[`Switching between the open and closed state manually Clicking the add button in the closed state switches to the open state 1`] = `
<AddInvestigator
  isOpen={false}
  potentialInvestigators={
    Array [
      Object {
        "id": "User id. 1",
        "name": "User nr. 1",
      },
      Object {
        "id": "User id. 2",
        "name": "User nr. 2",
      },
      Object {
        "id": "User id. 3",
        "name": "User nr. 3",
      },
      Object {
        "id": "User id. 4",
        "name": "User nr. 4",
      },
      Object {
        "id": "User id. 5",
        "name": "User nr. 5",
      },
      Object {
        "id": "User id. 6",
        "name": "User nr. 6",
      },
      Object {
        "id": "User id. 7",
        "name": "User nr. 7",
      },
      Object {
        "id": "User id. 8",
        "name": "User nr. 8",
      },
      Object {
        "id": "User id. 9",
        "name": "User nr. 9",
      },
      Object {
        "id": "User id. 10",
        "name": "User nr. 10",
      },
    ]
  }
>
  <div
    className="grid-content collapse shrink"
  >
    <form
      onSubmit={[Function]}
    >
      <span
        className="inline-label"
      >
        <AutocompleteInput
          alwaysRender={false}
          inputProps={
            Object {
              "autoFocus": true,
            }
          }
          onSelectionChange={[Function]}
          openOnFocus={false}
          options={
            Array [
              Object {
                "label": "User nr. 1",
                "value": "User id. 1",
              },
              Object {
                "label": "User nr. 2",
                "value": "User id. 2",
              },
              Object {
                "label": "User nr. 3",
                "value": "User id. 3",
              },
              Object {
                "label": "User nr. 4",
                "value": "User id. 4",
              },
              Object {
                "label": "User nr. 5",
                "value": "User id. 5",
              },
              Object {
                "label": "User nr. 6",
                "value": "User id. 6",
              },
              Object {
                "label": "User nr. 7",
                "value": "User id. 7",
              },
              Object {
                "label": "User nr. 8",
                "value": "User id. 8",
              },
              Object {
                "label": "User nr. 9",
                "value": "User id. 9",
              },
              Object {
                "label": "User nr. 10",
                "value": "User id. 10",
              },
            ]
          }
          placeholder="Add investigator ..."
        >
          <div
            className="autocomplete-override"
          >
            <Autocomplete
              autoHighlight={true}
              getItemValue={[Function]}
              inputProps={
                Object {
                  "autoFocus": true,
                }
              }
              isItemSelectable={[Function]}
              items={Array []}
              menuStyle={
                Object {
                  "background": "rgba(255, 255, 255, 0.9)",
                  "borderRadius": "3px",
                  "boxShadow": "0 2px 12px rgba(0, 0, 0, 0.1)",
                  "fontSize": "90%",
                  "maxHeight": "50%",
                  "overflow": "auto",
                  "padding": "2px 0",
                  "position": "fixed",
                }
              }
              onChange={[Function]}
              onMenuVisibilityChange={[Function]}
              onSelect={[Function]}
              renderInput={[Function]}
              renderItem={[Function]}
              renderMenu={[Function]}
              selectOnBlur={false}
              shouldItemRender={[Function]}
              value=""
              wrapperProps={Object {}}
              wrapperStyle={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <div
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-autocomplete="list"
                  aria-expanded={false}
                  autoComplete="off"
                  autoFocus={true}
                  className="autocomplete-input margin-none"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Add investigator ..."
                  role="combobox"
                  type="text"
                  value=""
                />
              </div>
            </Autocomplete>
          </div>
        </AutocompleteInput>
        <a
          className="button disabled add-investigator"
        >
          <i
            className="fa fa-plus margin-small-right"
          />
           Add
        </a>
        <a
          className="button secondary close-add-investigator"
          onClick={[Function]}
        >
          <i
            className="fa fa-times margin-small-right"
          />
          Cancel
        </a>
      </span>
    </form>
  </div>
</AddInvestigator>
`;

exports[`Switching between the open and closed state manually Clicking the cancel button in the open state switches to the closed state 1`] = `
<AddInvestigator
  isOpen={true}
  potentialInvestigators={
    Array [
      Object {
        "id": "User id. 1",
        "name": "User nr. 1",
      },
      Object {
        "id": "User id. 2",
        "name": "User nr. 2",
      },
      Object {
        "id": "User id. 3",
        "name": "User nr. 3",
      },
      Object {
        "id": "User id. 4",
        "name": "User nr. 4",
      },
      Object {
        "id": "User id. 5",
        "name": "User nr. 5",
      },
      Object {
        "id": "User id. 6",
        "name": "User nr. 6",
      },
      Object {
        "id": "User id. 7",
        "name": "User nr. 7",
      },
      Object {
        "id": "User id. 8",
        "name": "User nr. 8",
      },
      Object {
        "id": "User id. 9",
        "name": "User nr. 9",
      },
      Object {
        "id": "User id. 10",
        "name": "User nr. 10",
      },
    ]
  }
>
  <div
    className="grid-content collapse shrink"
  >
    <a
      className="button open-add-investigator"
      onClick={[Function]}
    >
      <i
        className="fa fa-plus"
      />
       Add investigator
    </a>
  </div>
</AddInvestigator>
`;
